pan/strings.pan

```diff
@@ -0,0 +1,23 @@
+declaration template pan/strings;
+
+
+@documentation{
+Convenience type specifying a string that must contain at least one character (anything other than an empty string).
+Intended for use in building more complex types.
+}
+type string_non_empty = string(1..);
+
+
+@documentation{
+Convenience type specifying a string that must contain at least one character and may not be contain any whitespace.
+Intended for use in building more complex types.
+}
+type string_non_whitespace = string_non_empty with !match(SELF, '\s');
+
+
+@documentation{
+Convenience type specifying a string that must contain at least one character, may contain whitespace, but may not
+start or end with whitespace.
+Intended for use in building more complex types.
+}
+type string_trimmed = string_non_empty with match(SELF, '^(?=\S).+(?<=\S)$');
```

Comment:
this regexp looks very complex, for something that is probably `^\S(.*\S)?$'

Reply:
Probably. I can't spend any more time on this now though.